geoportailv3/static/js/mymapsservice.js

```diff
@@ -0,0 +1,412 @@
+/**
+ * @fileoverview this file defines the mymaps webservice. this service
+ * interacts with the Geoportail MyMaps webservice and exposes functions that
+ * return objects representing maps and features.
+ */
+
+goog.provide('app.Mymaps');
+
+goog.require('app');
+goog.require('app.UserManager');
+
+
+/**
+ * @typedef {Array.<Object>}
+ */
+app.MapsResponse;
+
+
+
+/**
+ * @constructor
+ * @param {angular.$http} $http
+ * @param {string} mymapsMapsUrl URL to "mymaps" Maps service.
+ * @param {string} mymapsUrl URL to "mymaps" Features service.
+ * @param {app.DrawnFeatures} appDrawnFeatures Drawn features service.
+ * @param {app.StateManager} appStateManager
+ * @param {app.UserManager} appUserManager
+ * @ngInject
+ */
+app.Mymaps = function($http, mymapsMapsUrl, mymapsUrl, appDrawnFeatures,
+    appStateManager, appUserManager) {
+
+  /**
+   * @type {app.UserManager}
+   * @private
+   */
+  this.appUserManager_ = appUserManager;
+
+  /**
+   * @type {ol.Collection.<ol.Feature>}
+   * @private
+   */
+  this.drawnFeatures_ = appDrawnFeatures;
+
+  /**
+   * @type {app.StateManager}
+   * @private
+   */
+  this.stateManager_ = appStateManager;
+
+  /**
+   * @type {angular.$http}
+   * @private
+   */
+  this.$http_ = $http;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.mymapsMapsUrl_ = mymapsMapsUrl;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.mymapsFeaturesUrl_ = mymapsUrl + '/features/';
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.mymapsMapInfoUrl_ = mymapsUrl + '/map/';
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.mymapsDeleteFeatureUrl_ = mymapsUrl + '/delete_feature/';
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.mymapsDeleteMapUrl_ = mymapsUrl + '/delete/';
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.mymapsCreateMapUrl_ = mymapsUrl + '/create';
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.mymapsUpdateMapUrl_ = mymapsUrl + '/update/';
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.mymapsSaveFeatureUrl_ = mymapsUrl + '/save_feature/';
+
+  /**
+   * @type {?string}
+   * @private
+   */
+  this.mapId_ = null;
+
+  /**
+   * The currently displayed map title.
+   * @type {string}
+   * @export
+   */
+  this.mapTitle = '';
+
+  /**
+   * The currently displayed map title.
+   * @type {string}
+   * @export
+   */
+  this.mapOwner = '';
+
+  /**
+   * The currently displayed map description.
+   * @type {string}
+   * @export
+   */
+  this.mapDescription = '';
+
+  /**
+   * @type {ol.FeatureStyleFunction}
+   * @private
+   */
+  this.featureStyleFunction_ = app.DrawController.createStyleFunction();
```

Comment:
I don't think the `app.Mymaps` service should get the style function from the draw directive's controller. Services are standalone objects that directives may use. This is sort of going the other way around here.